ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from typing import Optional
+
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+
+
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: Optional[dict] = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self.request = request
+        self.payload = payload if payload else dict()
+        if not request:
```

Comment:
is this in case of no request, or no "request_name"? I am a bit confused about the logic.

Reply:
remove None opption from request and request_name.
Throw RequestNotFound if request name not in action_mapping.

+ Please add a test for this